bug: postinstall fails in pnpm-monorepo

The "postinstall" is specific to working on the package itself and not useful for consumers of the package.

Consider "prepare" instead? or maybe there is a better match: https://docs.npmjs.com/cli/v8/using-npm/scripts#life-cycle-scripts. You can also consider a custom bootstrap script for when you need to work on the package.

### Description

`postinstall` fails in a pnpm monorepo

### Expected outcome

That `normalize-diacritics` installs correctly

### Actual outcome
`<my-user>` and `<our-repo>` is  redacted information
```
...
node_modules/.pnpm/normalize-diacritics@3.0.11/node_modules/normalize-diacritics: Running postinstall script, failed in 481ms
.../node_modules/normalize-diacritics postinstall$ sh postinstall.sh
│ npm ERR! code ENOENT
│ npm ERR! syscall lstat
│ npm ERR! path /home/<my-user>/work/<our-repo>/lib
│ npm ERR! errno -2
│ npm ERR! enoent ENOENT: no such file or directory, lstat '/home/<my-user>/work/<our-repo>/lib'
│ npm ERR! enoent This is related to npm not being able to find a file.
│ npm ERR! enoent
│ npm ERR! A complete log of this run can be found in:
│ npm ERR!     /home/<my-user>/.npm/_logs/2022-11-08T09_59_58_504Z-debug-0.log
...
```

The line that triggers the error:
`postinstall.sh`
```
npx -y -- zx@latest ./postinstall.mjs
```

### Steps to reproduce
We run (in our monorepo)
`<repo>/package.json`
```
    "workspace--install": "git config  url.\"ssh://git@\".insteadOf git:// && pnpm recursive install --link-workspace-packages",
```

### Versions of affected system
* package: `normalize-diacritics@3.0.11`
* `pnpm --version` => 7.14.0
* `nix --version` => nix (Nix) 2.11.1
* platform: `Linux <my-hostname> 5.15.0-52-generic #58-Ubuntu SMP Thu Oct 13 08:03:55 UTC 2022 x86_64 GNU/Linux`

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -1,16 +1,11 @@
 name: ci
 
 on:
-  push:
-    branches:
-      - '*'
-    tags-ignore:
-      - '*'
   pull_request:
     branches:
-      - '*'
-    tags-ignore:
-      - '*'
+      - '**'
+    paths-ignore:
+      - '**/*.md'
 
 concurrency:
       group: ${{ github.workflow }}-${{ github.ref || github.run_id }}
@@ -37,7 +32,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        target: [14.x, 16.x, 18.x]
+        target: [16.x, 18.x]
 
     steps:
       - name: Checkout
@@ -87,7 +82,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        target: [14.x, 16.x, 18.x]
+        target: [16.x, 18.x]
 
     steps:
       - name: Checkout
@@ -156,7 +151,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        target: [14.x, 16.x, 18.x]
+        target: [16.x, 18.x]
 
     steps:
       - name: Checkout

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -69,6 +69,7 @@ jobs:
 
       - name: Install dependencies
         run: |
+          npm pkg delete scripts.postinstall
           npm ci
           sh $(npm root)/@reallyland/tools/list-npx-cache.sh
 
@@ -123,6 +124,7 @@ jobs:
 
       - name: Install dependencies
         run: |
+          npm pkg delete scripts.postinstall
           npm ci
           sh $(npm root)/@reallyland/tools/list-npx-cache.sh
 
@@ -186,6 +188,7 @@ jobs:
 
       - name: Install dependencies
         run: |
+          npm pkg delete scripts.postinstall
           npm ci
           sh $(npm root)/@reallyland/tools/list-npx-cache.sh
 

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -63,7 +63,20 @@ jobs:
 
       - name: Install dependencies
         run: |
+          npm pkg delete scripts.postinstall
           npm ci
+
+      - name: Lint
+        run: |
+          npm run lint:build
+
+      - name: Build
+        run: |
+          npm run build
+
+      - name: Cleanup package.json
+        run: |
+          npm pkg delete devDependencies nano-staged scripts
 
       - name: Publish to npm (Prerelease)
         if: contains(github.ref, '-rc') == true || contains(github.ref, '-') == true

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -2,10 +2,8 @@ name: publish
 
 on:
   push:
-    branches-ignore:
-      - '*'
     tags:
-      - '*'
+      - 'v**'
 
 defaults:
   run:
@@ -25,7 +23,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        target: [16.x]
+        target: [18.x]
 
     steps:
       - name: Checkout

--- a/package.json
+++ b/package.json
@@ -67,9 +67,7 @@
     "lint-commit": "sh $(npm root)/@reallyland/tools/lint-commit.sh",
     "lint:build": "sh $(npm root)/@reallyland/tools/lint-build.sh -c $(npm root)/@reallyland/tools/browser/.build.eslintrc.json",
     "pre-commit": "sh $(npm root)/@reallyland/tools/pre-commit.sh",
-    "prepublishOnly": "npm run lint:build && npm run build",
     "test": "vitest --coverage",
-    "ts": "tsc",
     "version": "sh $(npm root)/@reallyland/tools/generate-changelogs.sh && git add *CHANGELOG.md"
   },
   "dependencies": {

